### PR TITLE
update to latest proofs code, use new fr32 pad_reader, don't use multistore for lambda

### PR DESCRIPTION
**Pull request by rvagg**
Avoids the need to use the custom multistore which I don't have high confidence in, uses the streaming fr32 padding reader which shaves off ~1G from a ~<=1G input. Gets the Lambda version closer to rust-fil-proofs with less jank between. We're still injecting a `VecStore` backed `MerkleTree` which rust-fil-proofs doesn't use but the reimplementation of that is fairly trivial and mostly copypasta.

### CODE REVIEW

File: src/bin/lambda.rs [hunk 1]
```diff
@@ -69,7 +69,7 @@ fn commp_handler(request: CommPRequest, _c: Context) -> Result<CommPResponse, Ha
 
     let mut stream = result.body.unwrap().into_blocking_read();
 
-    let commp = commp::generate_commp_storage_proofs_mem(&mut stream, size, true).unwrap();
+    let commp = commp::generate_commp_storage_proofs_mem(&mut stream, size, false).unwrap();
```
Comment (rvagg) [inline src/bin/lambda.rs:72]: this is the key piece that switches the lambda from using the janky MultiStore to a plain VecStore one

**Comment by rvagg** [comment]
With a test file, here's the memory & disk usage for the calculation using VecStore and fr32 pad_reader:

```
	Size: 843.58 Mb
	Padded Size: 1,016.0 Mb
	Piece Size: 1,024.0 Mb
	CommP ee485c51deeeae03780547bb97bafcd0fc45872320237b0f697cbeac326ffb15
Took ~43 seconds
Peak disk 0 Mb
Peak mem 2598 Mb
Peak mem RSS 1986 Mb
```

And for using MultiStore and fr32 pad_reader:

```
	Size: 843.58 Mb
	Padded Size: 1,016.0 Mb
	Piece Size: 1,024.0 Mb
	CommP ee485c51deeeae03780547bb97bafcd0fc45872320237b0f697cbeac326ffb15
Took ~42 seconds
Peak disk 384 Mb
Peak mem 2276 Mb
Peak mem RSS 1635 Mb
```

Timing isn't so interesting, this is on a memory tmpfs so it's all RAM anyway.

File: src/lib.rs [hunk 2]
```diff
@@ -155,21 +171,22 @@ pub fn generate_commp_filecoin_proofs<R: Sized + io::Read>(
     })
 }
 
+/*
```
Comment (vmx) [inline src/lib.rs:183]: Is there any reason to keep that code? Why not just removing it (and all references), we still have it in the git history.

Comment (rvagg) [reply src/lib.rs:183]: Yeah, good point, I meant to do a bit more with dead code cleaning here. But, the point of this library is twofold - making a Lambda and exploring the methods to make this happen with the various filecoin_proofs APIs. We only really need one of the codepaths in lib.rs, making 3 others redundant. But they're there for the commandline just to test, experiment and demonstrate them!

So I've reenabled this section after updating it, cleaned some dead code and added docs at the head of each of the main functions.

File: src/lib.rs [hunk 3]
```diff
@@ -124,28 +143,35 @@ fn local_multistore_generate_piece_commitment_bytes_from_source<H: Hasher, R: Si
...
-    pad_reader
+    base2_pad_reader
 }
 
+/*
+ * How you'd do it through filecoin_proofs::generate_piece_commitment()
+ * which would be the more "correct" way to call in because most things
+ * are done for you.
+ */
 #[allow(dead_code)]
```
Comment (vmx) [inline src/lib.rs]: Please remove those `#[allow(dead_code)]` attributes, they are quite misleading (that's also OK to be done in a separate commit).

**Review by vmx — APPROVED** [review]

**Comment by rvagg** [comment]
all good and updated to match https://github.com/filecoin-project/rust-fil-proofs/pull/1063 assuming it'll land

**Review by vmx — APPROVED** [review]
